cmd/yokecd/main.go

```diff
@@ -112,15 +115,45 @@ func run(ctx context.Context, cfg Config) (err error) {
 		return fmt.Errorf("failed to execute flight wasm: %w", err)
 	}
 
-	return EncodeResources(json.NewEncoder(os.Stdout), data)
-}
+	stages, err := func(data []byte) (internal.Stages, error) {
+		stages, err := internal.ParseStages(data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse output into valid flight output: %w", err)
+		}
+		if len(stages) < 2 {
+			return stages, nil
+		}
+		for _, resource := range stages.Flatten() {
+			if annotations := resource.GetAnnotations(); annotations != nil {
+				if _, exists := annotations[annotationArgocdSyncWave]; exists {
+					return stages, nil
+				}
+			}
+		}
 
-func EncodeResources(out *json.Encoder, data []byte) error {
-	var resources internal.List[*unstructured.Unstructured]
-	if err := yaml.Unmarshal(data, &resources); err != nil {
-		return fmt.Errorf("failed to unmarshal executed flight data: %w", err)
+		for i, stage := range stages {
+			for _, resource := range stage {
+				annotations := resource.GetAnnotations()
+				if annotations == nil {
+					annotations = make(map[string]string)
+				}
+				annotations[annotationArgocdSyncWave] = fmt.Sprint(i)
+				resource.SetAnnotations(annotations)
+			}
+		}
+
+		return stages, nil
+	}(data)
+	if err != nil {
+		return err
 	}
 
+	internal.AddYokeMetadata(stages.Flatten(), cfg.Application.Name, cfg.Application.Namespace, "yokecd")
```

Comment:
I'm wondering if the manager has any significance and if it should instead be yoke or if yokecd is better

Reply:
Yokecd is good!